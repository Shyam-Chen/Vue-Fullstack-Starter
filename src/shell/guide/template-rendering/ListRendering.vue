<template>
  <v-layout column class="my-1">
    <div class="title mb-2">List Rendering</div>

    <ul class="pl-3 mb-1">
      <li v-for="(item, index) in listArray" :key="index">
        {{ item.message }}
      </li>
    </ul>

    <ul class="pl-3 mt-1">
      <li v-for="(value, index) in listObject" :key="index">
        {{ value }}
      </li>
    </ul>
  </v-layout>
</template>

<script>
// @flow

type Data = {
  listArray: Array<{ message: string }>,
  listObject: { foo: string, bar: string, baz: string },
};

export default {
  data(): Data {
    return {
      listArray: [{ message: 'Foo' }, { message: 'Bar' }, { message: 'Baz' }],
      listObject: {
        foo: 'Foo',
        bar: 'Bar',
        baz: 'Baz',
      },
    };
  },
};
</script>
