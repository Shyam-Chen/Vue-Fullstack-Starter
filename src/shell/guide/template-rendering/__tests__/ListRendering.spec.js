import { shallowMount } from '@vue/test-utils';

import ListRendering from '../ListRendering';

describe('ListRendering', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(ListRendering);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
