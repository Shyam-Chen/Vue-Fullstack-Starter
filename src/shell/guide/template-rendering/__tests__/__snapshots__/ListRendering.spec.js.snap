// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRendering should render an initial component 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">List Rendering</div>
  <ul class="pl-3 mb-1">
    <li>
      Foo
    </li>
    <li>
      Bar
    </li>
    <li>
      Baz
    </li>
  </ul>
  <ul class="pl-3 mt-1">
    <li>
      Foo
    </li>
    <li>
      Bar
    </li>
    <li>
      Baz
    </li>
  </ul>
</v-layout-stub>
`;
